Description: Fix libcrypto dynamic loading by reordering library filenames
 Load libcrypto by the development *.so symlink only as the last resort.
Origin: upstream, https://github.com/rhash/RHash/commit/633c7dd4d30d8d6f2cabb671e96e4b5df898a73d
Last-Update: 2022-06-28
Forwarded: not-needed

diff --git a/librhash/plug_openssl.c b/librhash/plug_openssl.c
--- a/librhash/plug_openssl.c
+++ b/librhash/plug_openssl.c
@@ -224,12 +224,12 @@ static int load_openssl_runtime(void)
 	SetErrorMode(oldErrorMode); /* restore error mode */
 #else
 	static const char* libNames[] = {
-		"libcrypto.so",
 		"libcrypto.so.3",
 		"libcrypto.so.1.1",
 		"libcrypto.so.1.0.2",
 		"libcrypto.so.1.0.0",
 		"libcrypto.so.0.9.8",
+		"libcrypto.so",
 	};
 	void* handle = 0;
 	size_t i;
